packages/trace-viewer/src/sw.ts

```diff
@@ -35,12 +35,19 @@ const loadedTraces = new Map<string, { traceModel: TraceModel, snapshotServer: S
 
 const clientIdToTraceUrls = new MultiMap<string, string>();
 
+class PrettyFetchError extends Error { }
+
+function validateZipEntry(entry: import('@zip.js/zip.js').Entry) {
+  if (entry.filename === 'index.html')
+    throw new PrettyFetchError('Could not load trace. Did you upload a Playwright HTML report instead? Make sure to extract the archive first and then double-click the index.html file or put it on a web server.');
+}
+
 async function loadTrace(trace: string, clientId: string, progress: (done: number, total: number) => void): Promise<TraceModel> {
   const entry = loadedTraces.get(trace);
   clientIdToTraceUrls.set(clientId, trace);
   if (entry)
     return entry.traceModel;
-  const traceModel = new TraceModel();
+  const traceModel = new TraceModel(validateZipEntry);
```

Comment:
I'd do the following:
- Make `TraceModel.load` throw a specific error when `traceEntry` is not found.
- Catch that error, search for the `index.html` entry in `traceModel` and provide a helpful error message.

That said, current implementation is ok, if it wouldn't be failing for valid trace files that also  have a stray `index.html` file inside.